llama_stack/providers/remote/inference/watsonx/watsonx.py

```diff
@@ -4,246 +4,115 @@
+
+    # get_api_key = LiteLLMOpenAIMixin.get_api_key
+
+    def get_base_url(self) -> str:
+        return self.config.url
+
+    async def initialize(self):
+        await super().initialize()
+
+    async def shutdown(self):
+        await super().shutdown()
+
+    async def _get_params(self, request: ChatCompletionRequest) -> dict[str, Any]:
+        # Get base parameters from parent
+        params = await super()._get_params(request)
+
+        # Add watsonx.ai specific parameters
+        params["project_id"] = self.config.project_id
+        params["time_limit"] = self.config.timeout
+        return params
+
+    async def check_model_availability(self, model):
+        return True
+
+    async def list_models(self) -> list[Model] | None:
+        models = []
+        for model_spec in self._get_model_specs():
+            functions = [f['id'] for f in model_spec.get("functions", [])]
+            # Format: {"embedding_dimension": 1536, "context_length": 8192}
+
+            # Example of an embedding model:
+            # {'model_id': 'ibm/granite-embedding-278m-multilingual',
+            # 'label': 'granite-embedding-278m-multilingual', 
+            # 'model_limits': {'max_sequence_length': 512, 'embedding_dimension': 768},
+            # ... 
+            if "embedding" in functions:
+                embedding_dimension = model_spec["model_limits"]["embedding_dimension"]
+                context_length = model_spec["model_limits"]["max_sequence_length"]
+                embedding_metadata = {
+                    "embedding_dimension": embedding_dimension,
+                    "context_length": context_length,
+                }
+                models.append(
+                    Model(
+                        identifier=model_spec["model_id"],
+                        provider_resource_id=f"{self.__provider_id__}/{model_spec['model_id']}",
+                        provider_id=self.__provider_id__,
+                        metadata=embedding_metadata,
+                        model_type=ModelType.embedding,
+                    )
+                )
+            if "text_chat" in functions:
+                models.append(
+                    Model(
+                        identifier=model_spec["model_id"],
+                        provider_resource_id=f"{self.__provider_id__}/{model_spec['model_id']}",
+                        provider_id=self.__provider_id__,
+                        metadata={},
+                        model_type=ModelType.llm,
+                    )
+                )
+        return models
+
+    # LiteLLM provides methods to list models for many providers, but not for watsonx.ai.
+    # So we need to implement our own method to list models by calling the watsonx.ai API.
+    def _get_model_specs(self) -> list[dict[str, Any]]:
+        """
+        Retrieves foundation model specifications from the watsonx.ai API.
+        """
+        url = f"{self.config.url}/ml/v1/foundation_model_specs?version=2023-10-25"
```

Comment:
```suggestion
        url = f"{self.config.url}/ml/v1/foundation_model_specs?version=2025-09-03"
```

Can we use a more recent version to get more models? We might want to expose a config flag for this?

Reply:
I meant to put in a comment here explaining why this is hard-coded, but I forgot.  I will include that comment in the next update and move to the latest API date version (2024-03-14).  In general we don't want to have a config flag for API date versions, because they only change when there is a breaking change to the API, so we shouldn't assume the code will work as implemented with a future API date version.  In any event, this API returns all the currently available models regardless of what API date version is used.